source/extensions/filters/common/rbac/engine_impl.cc

```diff
@@ -26,19 +26,27 @@ RoleBasedAccessControlEngineImpl::RoleBasedAccessControlEngineImpl(
     : RoleBasedAccessControlEngineImpl(per_route_config.rbac(), per_route_config.disabled()) {}
 
 bool RoleBasedAccessControlEngineImpl::allowed(const Network::Connection& connection,
-                                               const Envoy::Http::HeaderMap& headers) const {
+                                               const Envoy::Http::HeaderMap& headers,
+                                               bool& darklaunch_allowed) const {
   if (engine_disabled_) {
     return true;
   }
 
   bool matched = false;
+  bool darklaunch_matched_ = false;
   for (const auto& policy : policies_) {
     if (policy.matches(connection, headers)) {
-      matched = true;
-      break;
+      if (policy.mode() == envoy::config::rbac::v2alpha::PolicyEnforcementMode::POLICY_ENFORCEMENT_ENFORCED) {
```

Comment:
I'm not follow this logic. Can we keep this function as is, and in the filter, simply check if poicy.isDarklaunch() (or engine.isDarklaunch()) to decide if it should reject request or simply log result?